.github/workflows/publish.yml

```diff
@@ -31,6 +31,12 @@ jobs:
       - uses: actions/checkout@v4
       # Setup Dart SDK with JWT token
       - uses: dart-lang/setup-dart@0a8a0fc875eb934c15d08629302413c671d3f672
+      # Download flutter SDK - needed for publishing Flutter packages. Can also
+      # publish pure Dart packages.
+      #
+      # This will add the flutter/bin/dart executable to path, that sets up
+      # FLUTTER_ROOT.
+      - uses: flutter-actions/setup-flutter@d030cb603380106494f72d65a7e52462f380781f
```

Comment:
The `dart` binary from here will shadow the one installed from the previous step (`dart-lang/setup-dart`).

Reply:
And I don't know if this is an improvement - I haven't thought through the use cases - but we could parameterize this workflow (`workflow_call` > `inputs`, above) to indicate which sdk should be set up here.

Reply:
I can't think of any downsides (just that the optics are a little strange esp. given that this workflow is hosted in the `dart-lang/setup-dart` repo).

No concerns w/ this however. I would remove the line above - `uses: dart-lang/setup-dart` as that'll become a no-op given the `uses: flutter-actions/setup-flutter` line.

Reply:
> The setup-dart step is still important because it configures the OIDC token.

Gotcha. Perhaps update the comments in this file? `'Set up the Dart SDK and provision the OIDC token used for publishing'`; `'Set up the Flutter SDK; use the dart command from here to facilitate publishing both dart and flutter packages'`.

Reply:
Done